pkg/apis/yunikorn.apache.org/v1alpha1/type.go

```diff
@@ -33,15 +34,42 @@ type Application struct {
 	Status ApplicationStatus `json:"status"`
 }
 
-//Spec part
+// Spec part
 
 type ApplicationSpec struct {
-	MinMember         int32  `json:"minMember"`
-	Queue             string `json:"queue"`
-	MaxPendingSeconds int32  `json:"maxPendingSeconds,omitempty"`
+	Policy    SchedulePolicy `json:"schedulingPolicy"`
+	Queue     string         `json:"queue"`
+	TaskGroup []Task         `json:"taskGroups"`
 }
 
-//Status part
+type SchedulePolicy struct {
+	Policy    SchedulingPolicy      `json:"name"`
+	Parameter map[PolicyParam]int32 `json:"parameter,omitempty"`
+}
+
+type SchedulingPolicy string
+
+const (
+	TryOnce    SchedulingPolicy = "TryOnce"
+	MaxRetry   SchedulingPolicy = "MaxRetry"
+	TryReserve SchedulingPolicy = "TryReserve"
+	TryPreempt SchedulingPolicy = "TryPreempt"
+)
+
+type PolicyParam string
+
+const (
+	Timeout     PolicyParam = "Timeout"
+	RetrySecond PolicyParam = "RetrySecond"
+)
+
+type Task struct {
+	Name        string                       `json:"taskName"`
```

Comment:
this should be `groupName`
as we are declaring multiple gang groups here

Reply:
done.